api/views.py

```diff
@@ -230,6 +231,36 @@ class DeviceApiPicture(generics.RetrieveDestroyAPIView):
     queryset = Picture.objects.all()
     serializer_class = PictureSerializer
 
+class DeviceApiPictureRotate(generics.RetrieveUpdateAPIView):
+    model = Picture
+    queryset = Picture.objects.all()
+    serializer_class = PictureSerializer
+
+    def patch(self, request, *args, **kwargs):
+        from PIL import Image
+        import StringIO
+        from django.core.files.uploadedfile import InMemoryUploadedFile
+
+
+        picture = get_object_or_404(Picture, pk=self.kwargs["pk"])
+        img = Image.open(picture.image)
+
+        #determine if orientation is left or right
+        if self.kwargs["orientation"] == "right":
+            img  = img.transpose(Image.ROTATE_270)
+        if self.kwargs["orientation"] == "left":
+            img  = img.transpose(Image.ROTATE_90)
+        img_io = StringIO.StringIO()
```

Comment:
Can't we just use:

    img.save(picture.image.file.name)
    img.close()

and be done with it? :)

There is no need to save the picture object, right?

Reply:
The problem is, that Django expects a file and not an image object, as far as I know

Reply:
This approach works and it's much easier to read.

Reply:
Ok it works, but why xD

Reply:
It directly manipulates the referenced file.

Reply:
I think I misunderstood the way the save() function works...